server/service/src/boajs/call_method.rs

```diff
@@ -2,93 +2,92 @@ use std::{path::Path, rc::Rc};
-    #[error("Failed to locate plugins namespace for plugin {0}")]
-    PluginNamespaceMissing(String),
-    #[error("Failed to locate plugin {0}")]
-    PluginMissing(String),
+    #[error("Failed to load JS module")]
+    LoadingModule,
+    #[error("Failed to locate export {0}")]
+    ExportMissing(String),
     #[error(transparent)]
     SerdeError(#[from] serde_json::Error),
 }
 
-impl PartialEq for BoaJsPluginError {
+impl PartialEq for BoaJsError {
     fn eq(&self, _: &Self) -> bool {
         unimplemented!()
     }
 }
 
-fn plugin_type_to_string(r#type: &PluginType) -> String {
-    serde_json::to_string(r#type).unwrap().replace("\"", "")
-}
-
-pub(crate) fn call_plugin<I, O>(
+pub(crate) fn call_method<I, O>(
     input: I,
-    r#type: &PluginType,
+    // A path to exported method, plugins export { plugins: { plugin_name }}, thus we look for vec!["plugins", "plugin_name"]
+    // reports export { convert_data } thus we look for vec!["convert_data"]
+    export_location: Vec<&str>,
     bundle: &Vec<u8>,
-) -> Result<O, BoaJsPluginError>
+) -> Result<O, BoaJsError>
 where
     I: Serialize,
     O: DeserializeOwned,
 {
-    use BoaJsPluginError as Error;
-    let type_string = plugin_type_to_string(r#type);
-
+    use BoaJsError as Error;
     // Initialise context with loader
     let loader = Rc::new(SimpleModuleLoader::new(Path::new("."))?);
     let mut context = &mut Context::builder().module_loader(loader.clone()).build()?;
 
     // Add plugin code as module
     let module = Module::parse(Source::from_bytes(bundle), None, context)?;
-    loader.insert(Path::new("plugin.mjs").to_path_buf(), module.clone());
+    loader.insert(Path::new("module.mjs").to_path_buf(), module.clone());
 
     // Wait for module to load
     let promise = module.load_link_evaluate(context);
     context.run_jobs();
     match promise.state() {
         PromiseState::Fulfilled(JsValue::Undefined) => {}
-        _ => return Err(Error::LoadingModule(type_string.clone())),
+        _ => return Err(Error::LoadingModule),
     }
 
     // TODO should these be bound as camel case ? Also for inputs and outputs ?
     methods::log::bind_method(context)?;
     methods::sql::bind_method(context)?;
     methods::sql_type::bind_method(context)?;
-    methods::get_store_preferences::bind_method(context)?;
     methods::get_plugin_data::bind_method(context)?;
+    methods::get_store_preferences::bind_method(context)?;
 
-    let namespace = module.namespace(context);
-    let plugins = namespace
-        .get(js_string!("plugins"), context)?
-        .as_object()
-        .cloned()
-        .ok_or_else(|| Error::PluginNamespaceMissing(type_string.clone()))?;
-
-    let key = js_string!(type_string.as_str());
-    let plugin = plugins
-        .get(key, context)?
-        .as_callable()
-        .cloned()
-        .ok_or_else(|| Error::PluginMissing(type_string.clone()))?;
+    let callable = find_callable_in_exports(context, module, export_location)?;
```

Comment:
this is the main change really, you'll notice that previously as_callable was used, which still returns a JsObject (but checks if it's callable, i thought it's ok to error out when we try to call non callable